lib/Common/Memory/HeapBlock.cpp

```diff
@@ -187,7 +187,7 @@ SmallHeapBlockT<TBlockAttributes>::ConstructorCommon(HeapBucket * bucket, ushort
     this->Init(objectSize, objectCount);
     Assert(heapBlockType < HeapBlock::HeapBlockType::SmallAllocBlockTypeCount + HeapBlock::HeapBlockType::MediumAllocBlockTypeCount);
     Assert(objectCount > 1 && objectCount == (this->GetPageCount() * AutoSystemInfo::PageSize) / objectSize);
-#ifdef RECYCLER_SLOW_CHECK_ENABLED
+#if defined(RECYCLER_SLOW_CHECK_ENABLED) || (ENABLE_CONCURRENT_GC && ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP)
```

Comment:
Doesn't ENABLE_ALLOCATIONS_DURING_CONCCURRENT_SWEEEP => ENABLE_CONCURRENT_GC? When would the allocations be enabled but not concurrent gc?

Reply:
You are right. I could only check for ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP. I will simplify this everywhere.

---
In reply to: [148167429](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148167429) [](ancestors = 148167429)